[CLI]: test failures in v0.15.0 tag

### Describe the bug

<!--- Description of the issue below  -->
I'm observing the following test failures when running the test suite from tag v0.15.0:
```
=================================== FAILURES ===================================
______________________________ test_library_root _______________________________
[gw8] linux -- Python 3.10.10 /nix/store/fdqpyj613dr0v1l1lrzqhzay7sk4xg87-python3-3.10.10/bin/python3.10

    def test_library_root():
        symbol_list = dir(wandb)
        symbol_public_set = {s for s in symbol_list if not s.startswith("_")}
        print("symbols", symbol_public_set)
        symbol_unknown = (
            symbol_public_set
            - SYMBOLS_ROOT_DATATYPES
            - SYMBOLS_ROOT_SDK
            - SYMBOLS_ROOT_OTHER
            - SYMBOLS_TYPING
            - SYMBOLS_SERVICE
            - SYMBOLS_ANALYTICS
        )
>       assert symbol_unknown == set()
E       AssertionError: assert {'testing'} == set()
E         Extra items in the left set:
E         'testing'
E         Full diff:
E         - set()
E         + {'testing'}

/build/source/tests/pytest_tests/unit_tests/test_library_public.py:162: AssertionError
----------------------------- Captured stdout call -----------------------------
symbols {'teardown', 'JoinedTable', 'termsetup', 'catboost', 'require', 'sync', 'restore', 'cli', 'vendor', 'AlertLevel', 'InternalApi', 'keras', 'tensorflow', 'wandb_sdk', 'old', 'filesync', 'START_TIME', 'helper', 'watch', 'apis', 'docker', 'plots', 'jupyter', 'proto', 'xgboost', 'login', 'Graph', 'viz', 'patched', 'errors', 'log', 'Classes', 'integration', 'unwatch', 'sdk', 'log_artifact', 'gym', 'util', 'setup', 'Histogram', 'define_metric', 'api', 'Html', 'testing', 'sagemaker_auth', 'PublicApi', 'Api', 'finish', 'termwarn', 'mark_preempting', 'use_artifact', 'agent', 'controller', 'plot_table', 'lightgbm', 'profiler', 'Settings', 'Plotly', 'env', 'init', 'attach', 'ensure_configured', 'Molecule', 'Artifact', 'Video', 'trigger', 'sklearn', 'wandb_torch', 'Error', 'analytics', 'set_trace', 'Table', 'Audio', 'data_types', 'join', 'wandb_agent', 'UsageError', 'termerror', 'Object3D', 'Optional', 'Config', 'save', 'wandb_lib', 'plot', 'CommError', 'termlog', 'visualize', 'agents', 'summary', 'alert', 'config', 'run', 'tensorboard', 'sweep', 'Image', 'sacred', 'wandb', 'load_ipython_extension'}
__________________________ test_buildx_not_installed ___________________________
[gw2] linux -- Python 3.10.10 /nix/store/fdqpyj613dr0v1l1lrzqhzay7sk4xg87-python3-3.10.10/bin/python3.10

runner = <click.testing.CliRunner object at 0x7ffea4cad840>

    @pytest.mark.skipif(
        platform.system() == "Windows" or platform.system() == "Darwin",
        reason="this test fails incorrectly on CI for Windows and MacOS",
    )
    @pytest.mark.usefixtures("mock_shell")
    def test_buildx_not_installed(runner):
        with runner.isolated_filesystem():
            assert is_buildx_installed() is False
>           assert wandb.docker._buildx_installed is False
E           AssertionError: assert None is False
E            +  where None = <module 'wandb.docker' from '/build/source/wandb/docker/__init__.py'>._buildx_installed
E            +    where <module 'wandb.docker' from '/build/source/wandb/docker/__init__.py'> = wandb.docker

/build/source/tests/pytest_tests/unit_tests/test_docker.py:24: AssertionError
=============================== warnings summary ===============================
...
=========================== short test summary info ============================
FAILED tests/pytest_tests/unit_tests/test_library_public.py::test_library_root - AssertionError: assert {'testing'} == set()
FAILED tests/pytest_tests/unit_tests/test_docker.py::test_buildx_not_installed - AssertionError: assert None is False
```

<!--- A minimal code snippet between the quotes below  -->
To reproduce:
1. Checkout https://github.com/nixos/nixpkgs/commit/5167d6f405c450221daf839c02393a81145490b5
2. Replace https://github.com/nixos/nixpkgs/blob/master/pkgs/development/python-modules/wandb/default.nix with https://gist.github.com/samuela/c0cc466888160b3123134fd6a39f5015.
3. Run `nix-build -A python3Packages.wandb`.

### Additional Files

_No response_

### Environment

WandB version: 0.15.0

OS: Ubuntu 20.04.6 LTS

Python version: Python 3.10.10

Versions of relevant libraries: n/a


### Additional Context

_No response_

These issues are also [captured by Hydra](https://hydra.nixos.org/eval/1794694?filter=wandb&compare=1794679&full=):

1. On both `aarch64-linux` and `x86_64-linux`, the following test seems to be flaky:
```
=================================== FAILURES ===================================
_________________________ test_artifacts_cache_cleanup _________________________
[gw1] linux -- Python 3.10.11 /nix/store/ij979wyplsigcl0ndqkniyr7jj84xq2x-python3-3.10.11/bin/python3.10

cache = <wandb.sdk.interface.artifacts.artifact_cache.ArtifactsCache object at 0xfffdd81ed060>

    def test_artifacts_cache_cleanup(cache):
        cache_root = os.path.join(cache._cache_dir, "obj", "md5")
    
        path_1 = os.path.join(cache_root, "aa")
        os.makedirs(path_1)
        with open(os.path.join(path_1, "aardvark"), "w") as f:
            f.truncate(5000)
            f.flush()
            os.fsync(f)
    
        path_2 = os.path.join(cache_root, "ab")
        os.makedirs(path_2)
        with open(os.path.join(path_2, "absolute"), "w") as f:
            f.truncate(2000)
            f.flush()
            os.fsync(f)
    
        path_3 = os.path.join(cache_root, "ac")
        os.makedirs(path_3)
        with open(os.path.join(path_3, "accelerate"), "w") as f:
            f.truncate(1000)
            f.flush()
            os.fsync(f)
    
        reclaimed_bytes = cache.cleanup(5000)
    
        # We should get rid of "aardvark" in this case
>       assert reclaimed_bytes == 5000
E       assert 8000 == 5000

/build/source/tests/pytest_tests/unit_tests/test_artifacts/test_storage.py:162: AssertionError
=========================== short test summary info ============================
FAILED tests/pytest_tests/unit_tests/test_artifacts/test_storage.py::test_artifacts_cache_cleanup - assert 8000 == 5000
```

2. On `x86_64-darwin`, the following test fails.
```
=================================== FAILURES ===================================
____________________ test_copy_or_overwrite_changed_no_copy ____________________
[gw0] darwin -- Python 3.10.11 /nix/store/v6f6y9mwmqj9v1jffiy4g96ili4pj5la-python3-3.10.11/bin/python3.10

tmp_path = PosixPath('/private/tmp/nix-build-python3.10-wandb-0.15.0.drv-0/pytest-of-_nixbld4/pytest-0/popen-gw0/test_copy_or_overwrite_changed1')

    def test_copy_or_overwrite_changed_no_copy(tmp_path):
        source_path = tmp_path / "new_file.txt"
        target_path = tmp_path / "target_file.txt"
    
        source_path.write_text("original")
        shutil.copy2(source_path, target_path)
    
        with patch("shutil.copy2") as copy2_mock:
            copy_or_overwrite_changed(source_path, target_path)
>           assert not copy2_mock.called
E           AssertionError: assert not True
E            +  where True = <MagicMock name='copy2' id='8781091920'>.called

/private/tmp/nix-build-python3.10-wandb-0.15.0.drv-0/source/tests/pytest_tests/unit_tests/test_lib/test_filesystem.py:92: AssertionError
=========================== short test summary info ============================
FAILED tests/pytest_tests/unit_tests/test_lib/test_filesystem.py::test_copy_or_overwrite_changed_no_copy - AssertionError: assert not True
```

Also, on `aarch64-darwin`, the following test fails.

```
=================================== FAILURES ===================================
________________ test_safe_copy_target_file_changes_during_copy ________________
[gw0] darwin -- Python 3.10.11 /nix/store/xmxc0xc03fqq8b19sxfxwyd8k71kaz5m-python3-3.10.11/bin/python3.10
tmp_path = PosixPath('/private/tmp/nix-build-python3.10-wandb-0.15.3.drv-0/pytest-of-_nixbld2/pytest-0/popen-gw0/test_safe_copy_target_file_cha0')
monkeypatch = <_pytest.monkeypatch.MonkeyPatch object at 0x2c24fe2f0>
    def test_safe_copy_target_file_changes_during_copy(tmp_path: Path, monkeypatch):
        source_path = tmp_path / "source.txt"
        target_path = tmp_path / "target.txt"
        source_content = "Source content 📝" * 1000
        changed_target_content = "Changed target content 🔀"
    
        source_path.write_text(source_content, encoding="utf-8")
    
        def repeatedly_write_content():
            end_time = time.time() + 1.0
            while time.time() < end_time:
                try:
                    target_path.write_text(changed_target_content, encoding="utf-8")
                except PermissionError:
                    pass
    
        def delayed_copy_with_pause(src, dst, *args, **kwargs):
            """Write a 4096 byte block at a time, pausing 0.1 seconds between writes."""
            # Windows often doesn't allow opening a file for writing while there is an open
            # file handle to it. To get around the PermissionError, we close the file
            # between writes, but also retry failed writes.
            pos = 0
            with open(src, "rb") as infile:
                for block in iter(lambda: infile.read(4096), b""):
                    success = False
                    while not success:
                        time.sleep(0.1)
                        try:
                            with open(dst, "wb") as outfile:
                                outfile.seek(pos)
                                outfile.write(block)
                            pos += len(block)
                            success = True
                        except PermissionError:
                            pass
    
        monkeypatch.setattr(shutil, "copy2", delayed_copy_with_pause)
    
        with ThreadPoolExecutor(max_workers=2) as executor:
            future = executor.submit(repeatedly_write_content)
            safe_copy(source_path, target_path)
            future.result()
    
        result_content = target_path.read_text("utf-8")
>       assert result_content == source_content or result_content == changed_target_content
E       AssertionError: assert 
=========================== short test summary info ============================
FAILED tests/pytest_tests/unit_tests/test_lib/test_filesystem.py::test_safe_copy_target_file_changes_during_copy - AssertionError
```

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -231,7 +231,6 @@ ignore_errors = false
 
 [[tool.mypy.overrides]]
 module = [
-    "wandb.testing.*",
     "wandb.integration.*",
     "wandb.apis.*",
     "wandb.proto.*",

--- a/tests/system_tests/conftest.py
+++ b/tests/system_tests/conftest.py
@@ -17,7 +17,8 @@ import requests
 import wandb
 import wandb.old.settings
 import wandb.util
-from wandb.testing.relay import (
+
+from .relay import (
     DeliberateHTTPError,
     InjectedResponse,
     RelayServer,

--- a/tests/system_tests/conftest.py
+++ b/tests/system_tests/conftest.py
@@ -1043,3 +1043,8 @@ def inject_graphql_response(base_url, user):
         )
 
     yield helper
+
+
+@pytest.fixture(scope="function")
+def tokenized_circular_pattern():
+    return TokenizedCircularPattern

--- a/tests/system_tests/relay.py
+++ b/tests/system_tests/relay.py
@@ -24,7 +24,6 @@ import flask
 import pandas as pd
 import requests
 import responses
-
 import wandb
 import wandb.util
 from wandb.sdk.lib.timer import Timer

--- a/tests/system_tests/test_artifacts/test_misc2.py
+++ b/tests/system_tests/test_artifacts/test_misc2.py
@@ -5,17 +5,18 @@ import pytest
 import wandb
 from wandb.sdk.artifacts.exceptions import ArtifactNotLoggedError
 
-from ..relay import TokenizedCircularPattern
 
-
-def test_artifact_log_with_network_error(user, relay_server, inject_graphql_response):
+def test_artifact_log_with_network_error(
+    user, relay_server, inject_graphql_response, tokenized_circular_pattern
+):
     create_artifact_response = inject_graphql_response(
         # request
         query_match_fn=lambda query, variables: query.startswith(
             "mutation CreateArtifact("
         ),
         application_pattern=(
-            TokenizedCircularPattern.APPLY_TOKEN + TokenizedCircularPattern.STOP_TOKEN
+            tokenized_circular_pattern.APPLY_TOKEN
+            + tokenized_circular_pattern.STOP_TOKEN
         ),
         # response
         status=500,

--- a/tests/system_tests/test_artifacts/test_misc2.py
+++ b/tests/system_tests/test_artifacts/test_misc2.py
@@ -4,7 +4,8 @@ import numpy as np
 import pytest
 import wandb
 from wandb.sdk.artifacts.exceptions import ArtifactNotLoggedError
-from wandb.testing.relay import TokenizedCircularPattern
+
+from ..relay import TokenizedCircularPattern
 
 
 def test_artifact_log_with_network_error(user, relay_server, inject_graphql_response):

--- a/tests/system_tests/test_core/test_file_stream.py
+++ b/tests/system_tests/test_core/test_file_stream.py
@@ -3,7 +3,7 @@ import re
 import pytest
 import wandb
 
-from ..relay import TokenizedCircularPattern
+# from ..relay import TokenizedCircularPattern
 
 
 def log_line_match_http_error(user, project, run_id, status_code):
@@ -43,6 +43,7 @@ def test_retryable_codes(
     relay_server,
     inject_file_stream_response,
     monkeypatch,
+    tokenized_circular_pattern,
 ):
     # turn on debug logs
     monkeypatch.setenv("WANDB_DEBUG", "true")
@@ -61,9 +62,9 @@ def test_retryable_codes(
             inject_file_stream_response(
                 run=run,
                 application_pattern=(
-                    TokenizedCircularPattern.APPLY_TOKEN
-                    + TokenizedCircularPattern.APPLY_TOKEN
-                    + TokenizedCircularPattern.STOP_TOKEN
+                    tokenized_circular_pattern.APPLY_TOKEN
+                    + tokenized_circular_pattern.APPLY_TOKEN
+                    + tokenized_circular_pattern.STOP_TOKEN
                 ),
                 status=status_code,
                 body="transient error",
@@ -103,6 +104,7 @@ def test_non_retryable_codes(
     relay_server,
     inject_file_stream_response,
     monkeypatch,
+    tokenized_circular_pattern,
 ):
     # turn on debug logs
     monkeypatch.setenv("WANDB_DEBUG", "true")
@@ -121,8 +123,8 @@ def test_non_retryable_codes(
             inject_file_stream_response(
                 run=run,
                 application_pattern=(
-                    TokenizedCircularPattern.APPLY_TOKEN
-                    + TokenizedCircularPattern.STOP_TOKEN
+                    tokenized_circular_pattern.APPLY_TOKEN
+                    + tokenized_circular_pattern.STOP_TOKEN
                 ),
                 status=status_code,
                 body="non-retryable error",
@@ -145,6 +147,7 @@ def test_connection_reset(
     relay_server,
     inject_file_stream_connection_reset,
     monkeypatch,
+    tokenized_circular_pattern,
 ):
     # turn on debug logs
     monkeypatch.setenv("WANDB_DEBUG", "true")
@@ -163,9 +166,9 @@ def test_connection_reset(
             inject_file_stream_connection_reset(
                 run=run,
                 application_pattern=(
-                    TokenizedCircularPattern.APPLY_TOKEN
-                    + TokenizedCircularPattern.APPLY_TOKEN
-                    + TokenizedCircularPattern.STOP_TOKEN
+                    tokenized_circular_pattern.APPLY_TOKEN
+                    + tokenized_circular_pattern.APPLY_TOKEN
+                    + tokenized_circular_pattern.STOP_TOKEN
                 ),
                 body=ConnectionResetError("Connection reset by peer"),
             )

--- a/tests/system_tests/test_core/test_file_stream.py
+++ b/tests/system_tests/test_core/test_file_stream.py
@@ -2,7 +2,8 @@ import re
 
 import pytest
 import wandb
-from wandb.testing.relay import TokenizedCircularPattern
+
+from ..relay import TokenizedCircularPattern
 
 
 def log_line_match_http_error(user, project, run_id, status_code):

--- a/tests/system_tests/test_core/test_file_stream.py
+++ b/tests/system_tests/test_core/test_file_stream.py
@@ -3,8 +3,6 @@ import re
 import pytest
 import wandb
 
-# from ..relay import TokenizedCircularPattern
-
 
 def log_line_match_http_error(user, project, run_id, status_code):
     pattern = (

